### ISSUE TITLE
[Bug]: Dockercompose buildpack on git repository mounting a file results in refering to undefined volume

### ISSUE THREAD

**Issue by carbontwelve** [coollabsio/coolify#2851]
### Description

I have a git repository that contains some config files I am attempting to mount within containers spun up via docker compose.

The repository source yaml looks like this:
```yaml
    volumes:
      - type: bind
        source: ./telegraf/etc/telegraf.conf
        target: /etc/telegraf/telegraf.conf
        read_only: true
```

Coolify is then turning that into:
```yaml
    volumes:
      - 'esc0ssw-/data/coolify/applications/esc0ssw/telegraf/etc/telegraf.conf:/etc/telegraf/telegraf.conf:ro'
```

This results in a failed deployment with the following error:

> service "telegraf" refers to undefined volume esc0ssw-/data/coolify/applications/esc0ssw/telegraf/etc/telegraf.conf: invalid compose project

### Minimal Reproduction (if possible, example repository)

Create a new public git deployiment from https://github.com/photogabble/coolify-dashboard and select docker compose buildpack.

### Exception or Error

```
[2024-Jul-16 23:42:32.314339] Starting deployment of photogabble/coolify-dashboard:main-so8ggkc to Moose.
[2024-Jul-16 23:42:32.410704] Preparing container with helper image: ghcr.io/coollabsio/coolify-helper:latest.
[2024-Jul-16 23:42:32.468910]

[COMMAND] docker rm -f p4csw8s
[OUTPUT]
Error response from daemon: No such container: p4csw8s

[2024-Jul-16 23:42:32.609082]

[COMMAND] docker run -d --network coolify --name p4csw8s --rm -v /var/run/docker.sock:/var/run/docker.sock ghcr.io/coollabsio/coolify-helper:latest
[OUTPUT]
a090e21bdd0ada9b1ba14139d789f55c4ead3ec866856ee506d69afaebf9f0d0

[2024-Jul-16 23:42:33.685999]

[COMMAND] docker exec p4csw8s bash -c 'GIT_SSH_COMMAND="ssh -o ConnectTimeout=30 -p 22 -o Port=22 -o LogLevel=ERROR -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" git ls-remote https://github.com/photogabble/coolify-dashboard main'
[OUTPUT]
00d8feaf99cff3c1c9ef678a8aff1df1cdbb0548 refs/heads/main

[2024-Jul-16 23:42:33.699823] ----------------------------------------
[2024-Jul-16 23:42:33.701739] Importing photogabble/coolify-dashboard:main (commit sha HEAD) to /artifacts/p4csw8s.
[2024-Jul-16 23:42:33.824361]

[COMMAND] docker exec p4csw8s bash -c 'git clone -b "main" https://github.com/photogabble/coolify-dashboard /artifacts/p4csw8s && cd /artifacts/p4csw8s && sed -i "s#git@\(.*\):#https://\1/#g" /artifacts/p4csw8s/.gitmodules || true && cd /artifacts/p4csw8s && GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" git submodule update --init --recursive && cd /artifacts/p4csw8s && GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" git lfs pull'
[OUTPUT]
Cloning into '/artifacts/p4csw8s'...

[2024-Jul-16 23:42:34.365204]

[COMMAND] docker exec p4csw8s bash -c 'git clone -b "main" https://github.com/photogabble/coolify-dashboard /artifacts/p4csw8s && cd /artifacts/p4csw8s && sed -i "s#git@\(.*\):#https://\1/#g" /artifacts/p4csw8s/.gitmodules || true && cd /artifacts/p4csw8s && GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" git submodule update --init --recursive && cd /artifacts/p4csw8s && GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" git lfs pull'
[OUTPUT]
sed: /artifacts/p4csw8s/.gitmodules: No such file or directory

[2024-Jul-16 23:42:34.708424]

[COMMAND] docker exec p4csw8s bash -c 'cd /artifacts/p4csw8s && git log -1 00d8feaf99cff3c1c9ef678a8aff1df1cdbb0548 --pretty=%B'
[OUTPUT]
chore: mount telegraf.conf as file

[2024-Jul-16 23:42:35.840396] Pulling & building required images.
[2024-Jul-16 23:42:36.021392]

[COMMAND] docker exec p4csw8s bash -c 'SOURCE_COMMIT=00d8feaf99cff3c1c9ef678a8aff1df1cdbb0548 COOLIFY_BRANCH=main docker compose --env-file /artifacts/p4csw8s/.env --project-name esc0ssw --project-directory /artifacts/p4csw8s -f /artifacts/p4csw8s/docker-compose.yaml build --pull'
[OUTPUT]
service "telegraf" refers to undefined volume esc0ssw-/data/coolify/applications/esc0ssw/telegraf/etc/telegraf.conf: invalid compose project

[2024-Jul-16 23:42:36.051236] Oops something is not okay, are you okay? 😢

[2024-Jul-16 23:42:36.053245] service "telegraf" refers to undefined volume esc0ssw-/data/coolify/applications/esc0ssw/telegraf/etc/telegraf.conf: invalid compose project
```

### Version

v4.0.0-beta.314

**Comment by andrasbacsai** [comment]
This will be fixed in the upcoming version.

**Comment by carbontwelve** [comment]
@andrasbacsai wow that was quick! I can confirm that this is fixed in v4.0.0-beta.315, I'm however now being hindered by #1996.
